### PR TITLE
Standardize header guards

### PR DESCRIPTION
**Pull request by jonsimantov**
Standardize header guards, and turn on lint warnings for bad header guard names and line length, which were previously disabled. Includes a small fix to replace " with &amp;quot; in the lint output, for HTML compatibility.

This also turns off the "include subdirectory" and "whitespace/indent" lint warnings as they are extraneous.

### CODE REVIEW

**Review by DellaBitta — APPROVED** [review]

File: app/src/include/firebase/app.h [hunk 1]
```diff
@@ -14,8 +14,8 @@
  * limitations under the License.
  */
 
-#ifndef FIREBASE_APP_CLIENT_CPP_SRC_INCLUDE_FIREBASE_APP_H_
-#define FIREBASE_APP_CLIENT_CPP_SRC_INCLUDE_FIREBASE_APP_H_
+#ifndef FIREBASE_APP_SRC_INCLUDE_FIREBASE_APP_H_
+#define FIREBASE_APP_SRC_INCLUDE_FIREBASE_APP_H_
```
Comment (jonsimantov) [inline app/src/include/firebase/app.h:19]: 📝 Documentation warning:warning: argument 'confirg' of command @param is not found in the argument list of firebase::AppOptions::LoadFromJsonConfig(const char *config, AppOptions *options=nullptr)`<hidden value="hidden-doxygen-comment-tag"></hidden>

File: app/src/include/firebase/app.h [hunk 2]
```diff
@@ -14,8 +14,8 @@
...
 
 #include "firebase/internal/platform.h"
```
Comment (jonsimantov) [inline app/src/include/firebase/app.h:21]: 📝 Documentation warning:warning: The following parameter of firebase::AppOptions::LoadFromJsonConfig(const char *config, AppOptions *options=nullptr) is not documented:
  parameter 'config'`<hidden value="hidden-doxygen-comment-tag"></hidden>

File: app/src/include/firebase/app.h (hunk 1 above)
Comment (jonsimantov) [inline app/src/include/firebase/app.h:19]: 📝 Documentation warning: `warning: argument 'confirg' of command @param is not found in the argument list of firebase::AppOptions::LoadFromJsonConfig(const char *config, AppOptions *options=nullptr)`<hidden value="hidden-doxygen-comment-tag"></hidden>

File: app/src/include/firebase/app.h (hunk 2 above)
Comment (jonsimantov) [inline app/src/include/firebase/app.h:21]: 📝 Documentation warning: `warning: The following parameter of firebase::AppOptions::LoadFromJsonConfig(const char *config, AppOptions *options=nullptr) is not documented:
  parameter 'config'`<hidden value="hidden-doxygen-comment-tag"></hidden>

File: app/src/include/firebase/app.h (hunk 1 above)
Comment (jonsimantov) [inline app/src/include/firebase/app.h:19]: 📝 Documentation warning: ```warning: argument 'confirg' of command @param is not found in the argument list of firebase::AppOptions::LoadFromJsonConfig(const char *config, AppOptions *options=nullptr)```<hidden value="hidden-doxygen-comment-tag"></hidden>

File: app/src/include/firebase/app.h (hunk 2 above)
Comment (jonsimantov) [inline app/src/include/firebase/app.h:21]: 📝 Documentation warning: ```warning: The following parameter of firebase::AppOptions::LoadFromJsonConfig(const char *config, AppOptions *options=nullptr) is not documented:
  parameter 'config'```<hidden value="hidden-doxygen-comment-tag"></hidden>

File: app/src/include/firebase/app.h (hunk 1 above)
Comment (jonsimantov) [inline app/src/include/firebase/app.h:19]: 📝 Documentation warning: \
`warning: argument 'confirg' of command @param is not found in the argument list of firebase::AppOptions::LoadFromJsonConfig(const char *config, AppOptions *options=nullptr)`<hidden value="hidden-doxygen-comment-tag"></hidden>

File: app/src/include/firebase/app.h (hunk 2 above)
Comment (jonsimantov) [inline app/src/include/firebase/app.h:21]: 📝 Documentation warning: \
`warning: The following parameter of firebase::AppOptions::LoadFromJsonConfig(const char *config, AppOptions *options=nullptr) is not documented:`
`%s`  parameter 'config'<hidden value="hidden-doxygen-comment-tag"></hidden>

File: app/src/include/firebase/app.h (hunk 1 above)
Comment (jonsimantov) [inline app/src/include/firebase/app.h:19]: 📝 Documentation warning: \
`warning: argument 'confirg' of command @param is not found in the argument list of firebase::AppOptions::LoadFromJsonConfig(const char *config, AppOptions *options=nullptr)`<hidden value="hidden-doxygen-comment-tag"></hidden>

File: app/src/include/firebase/app.h (hunk 2 above)
Comment (jonsimantov) [inline app/src/include/firebase/app.h:21]: 📝 Documentation warning: \
`warning: The following parameter of firebase::AppOptions::LoadFromJsonConfig(const char *config, AppOptions *options=nullptr) is not documented:`
`  parameter 'config'`<hidden value="hidden-doxygen-comment-tag"></hidden>

File: app/src/include/firebase/app.h (hunk 1 above)
Comment (jonsimantov) [inline app/src/include/firebase/app.h:19]: 📝 Documentation issue: <br/>`warning: argument 'confirg' of command @param is not found in the argument list of firebase::AppOptions::LoadFromJsonConfig(const char *config, AppOptions *options=nullptr)`<hidden value="hidden-doxygen-comment-tag"></hidden>

File: app/src/include/firebase/app.h (hunk 2 above)
Comment (jonsimantov) [inline app/src/include/firebase/app.h:21]: 📝 Documentation issue: <br/>`warning: The following parameter of firebase::AppOptions::LoadFromJsonConfig(const char *config, AppOptions *options=nullptr) is not documented:`
`  parameter 'config'`<hidden value="hidden-doxygen-comment-tag"></hidden>

File: app/src/include/firebase/app.h (hunk 1 above)
Comment (jonsimantov) [inline app/src/include/firebase/app.h:19]: 📝 &nbsp;__Documentation issue:__ `warning: argument 'confirg' of command @param is not found in the argument list of firebase::AppOptions::LoadFromJsonConfig(const char *config, AppOptions *options=nullptr)`<hidden value="hidden-doxygen-comment-tag"></hidden>

File: app/src/include/firebase/app.h (hunk 2 above)
Comment (jonsimantov) [inline app/src/include/firebase/app.h:21]: 📝 &nbsp;__Documentation issue:__ `warning: The following parameter of firebase::AppOptions::LoadFromJsonConfig(const char *config, AppOptions *options=nullptr) is not documented:`
`  parameter 'config'`<hidden value="hidden-doxygen-comment-tag"></hidden>

File: app/src/include/firebase/app.h (hunk 1 above)
Comment (jonsimantov) [inline app/src/include/firebase/app.h:19]: 📝 &nbsp;__Documentation issue:__ `warning: argument 'confirg' of command @param is not found in the argument list of firebase::AppOptions::LoadFromJsonConfig(const char *config, AppOptions *options=nullptr)`<hidden value="hidden-doxygen-comment-tag"></hidden>

File: app/src/include/firebase/app.h (hunk 2 above)
Comment (jonsimantov) [inline app/src/include/firebase/app.h:21]: 📝 &nbsp;__Documentation issue:__ `warning: The following parameter of firebase::AppOptions::LoadFromJsonConfig(const char *config, AppOptions *options=nullptr) is not documented:`
`  parameter 'config'`<hidden value="hidden-doxygen-comment-tag"></hidden>

File: app/src/include/firebase/app.h (hunk 1 above)
Comment (jonsimantov) [inline app/src/include/firebase/app.h:19]: 📝 &nbsp;__Documentation issue:__ `warning: argument 'confirg' of command @param is not found in the argument list of firebase::AppOptions::LoadFromJsonConfig(const char *config, AppOptions *options=nullptr)`<hidden value="hidden-doxygen-comment-tag"></hidden>

File: app/src/include/firebase/app.h (hunk 2 above)
Comment (jonsimantov) [inline app/src/include/firebase/app.h:21]: 📝 &nbsp;__Documentation issue:__ `warning: The following parameter of firebase::AppOptions::LoadFromJsonConfig(const char *config, AppOptions *options=nullptr) is not documented:`
`  parameter 'config'`<hidden value="hidden-doxygen-comment-tag"></hidden>

File: app/src/include/firebase/app.h (hunk 1 above)
Comment (jonsimantov) [inline app/src/include/firebase/app.h:19]: 📝 &nbsp;__Documentation issue:__ `warning: argument 'confirg' of command @param is not found in the argument list of firebase::AppOptions::LoadFromJsonConfig(const char *config, AppOptions *options=nullptr)`<hidden value="hidden-doxygen-comment-tag"></hidden>

File: app/src/include/firebase/app.h (hunk 2 above)
Comment (jonsimantov) [inline app/src/include/firebase/app.h:21]: 📝 &nbsp;__Documentation issue:__ [8 lines up] `warning: The following parameter of firebase::AppOptions::LoadFromJsonConfig(const char *config, AppOptions *options=nullptr) is not documented:`
`  parameter 'config'`<hidden value="hidden-doxygen-comment-tag"></hidden>

File: app/src/include/firebase/app.h (hunk 1 above)
Comment (jonsimantov) [inline app/src/include/firebase/app.h:19]: 📝 &nbsp;__Documentation issue:__ `warning: argument 'confirg' of command @param is not found in the argument list of firebase::AppOptions::LoadFromJsonConfig(const char *config, AppOptions *options=nullptr)`<hidden value="hidden-doxygen-comment-tag"></hidden>

File: app/src/include/firebase/app.h (hunk 2 above)
Comment (jonsimantov) [inline app/src/include/firebase/app.h:21]: 📝 &nbsp;__Documentation issue:__ [8 lines down] `warning: The following parameter of firebase::AppOptions::LoadFromJsonConfig(const char *config, AppOptions *options=nullptr) is not documented:`
`  parameter 'config'`<hidden value="hidden-doxygen-comment-tag"></hidden>

File: app/src/include/firebase/app.h (hunk 1 above)
Comment (jonsimantov) [inline app/src/include/firebase/app.h:19]: 📝 &nbsp;__Documentation issue:__ `warning: argument 'confirg' of command @param is not found in the argument list of firebase::AppOptions::LoadFromJsonConfig(const char *config, AppOptions *options=nullptr)`<hidden value="hidden-doxygen-comment-tag"></hidden>

File: app/src/include/firebase/app.h (hunk 2 above)
Comment (jonsimantov) [inline app/src/include/firebase/app.h:21]: 📝 &nbsp;__Documentation issue:__ `[8 lines down] ``warning: The following parameter of firebase::AppOptions::LoadFromJsonConfig(const char *config, AppOptions *options=nullptr) is not documented:`
`  parameter 'config'`<hidden value="hidden-doxygen-comment-tag"></hidden>

File: app/src/include/firebase/app.h (hunk 1 above)
Comment (jonsimantov) [inline app/src/include/firebase/app.h:19]: 📝 &nbsp;__Documentation issue:__ `warning: argument 'confirg' of command @param is not found in the argument list of firebase::AppOptions::LoadFromJsonConfig(const char *config, AppOptions *options=nullptr)`<hidden value="hidden-doxygen-comment-tag"></hidden>

File: app/src/include/firebase/app.h (hunk 2 above)
Comment (jonsimantov) [inline app/src/include/firebase/app.h:21]: 📝 &nbsp;__Documentation issue:__ `[8 lines down] warning: The following parameter of firebase::AppOptions::LoadFromJsonConfig(const char *config, AppOptions *options=nullptr) is not documented:`
`  parameter 'config'`<hidden value="hidden-doxygen-comment-tag"></hidden>

File: app/src/include/firebase/app.h (hunk 1 above)
Comment (jonsimantov) [inline app/src/include/firebase/app.h:19]: 📝  __Documentation issue:__ `warning: argument 'confirg' of command @param is not found in the argument list of firebase::AppOptions::LoadFromJsonConfig(const char *config, AppOptions *options=nullptr)`<hidden value="hidden-doxygen-comment-tag"></hidden>

File: app/src/include/firebase/app.h (hunk 2 above)
Comment (jonsimantov) [inline app/src/include/firebase/app.h:21]: 📝  __Documentation issue:__ `[8 lines down] warning: The following parameter of firebase::AppOptions::LoadFromJsonConfig(const char *config, AppOptions *options=nullptr) is not documented:`
`  parameter 'config'`<hidden value="hidden-doxygen-comment-tag"></hidden>

File: app/src/include/firebase/app.h (hunk 1 above)
Comment (jonsimantov) [inline app/src/include/firebase/app.h:19]: 📝  __Documentation issue:__ `warning: argument 'confirg' of command @param is not found in the argument list of firebase::AppOptions::LoadFromJsonConfig(const char *config, AppOptions *options=nullptr)`<hidden value="hidden-doxygen-comment-tag"></hidden>

File: app/src/include/firebase/app.h (hunk 2 above)
Comment (jonsimantov) [inline app/src/include/firebase/app.h:21]: 📝  __Documentation issue:__ `[8 lines down] warning: The following parameter of firebase::AppOptions::LoadFromJsonConfig(const char *config, AppOptions *options=nullptr) is not documented:`
`  parameter 'config'`<hidden value="hidden-doxygen-comment-tag"></hidden>

File: app/src/include/firebase/app.h (hunk 1 above)
Comment (jonsimantov) [inline app/src/include/firebase/app.h:19]: 📝  __Documentation issue:__ `warning: argument 'confirg' of command @param is not found in the argument list of firebase::AppOptions::LoadFromJsonConfig(const char *config, AppOptions *options=nullptr)`<hidden value="hidden-doxygen-comment-tag"></hidden>

File: app/src/include/firebase/app.h (hunk 2 above)
Comment (jonsimantov) [inline app/src/include/firebase/app.h:21]: 📝  __Documentation issue:__ `[8 lines down] warning: The following parameter of firebase::AppOptions::LoadFromJsonConfig(const char *config, AppOptions *options=nullptr) is not documented:`
`  parameter 'config'`<hidden value="hidden-doxygen-comment-tag"></hidden>

File: app/src/include/firebase/app.h (hunk 1 above)
Comment (jonsimantov) [inline app/src/include/firebase/app.h:19]: 📝  __Documentation issue:__ `warning: argument 'confirg' of command @param is not found in the argument list of firebase::AppOptions::LoadFromJsonConfig(const char *config, AppOptions *options=nullptr)`<hidden value="hidden-doxygen-comment-tag"></hidden>

File: app/src/include/firebase/app.h (hunk 2 above)
Comment (jonsimantov) [inline app/src/include/firebase/app.h:21]: 📝  __Documentation issue:__ `[8 lines down] warning: The following parameter of firebase::AppOptions::LoadFromJsonConfig(const char *config, AppOptions *options=nullptr) is not documented:`
`  parameter 'config'`<hidden value="hidden-doxygen-comment-tag"></hidden>

**Pull request merged**